packages/xl-ai/src/AIExtension.ts

```diff
@@ -0,0 +1,195 @@
+import {
+  BlockNoteEditor,
+  BlockNoteExtension,
+  UnreachableCaseError,
+} from "@blocknote/core";
+import { LanguageModel } from "ai";
+import { createStore } from "zustand/vanilla";
+import { PromptOrMessages, llm } from "./api";
+import { LLMRequestOptions } from "./api/streamTool/callLLMWithStreamTools";
+
+type AIPluginState = {
+  aiMenuBlockID: string | undefined;
+  aiMenuResponseStatus: "initial" | "generating" | "error" | "done";
+};
```

Comment:
What if this was a union, would setState be able to enforce this properly?

Like, what if it was:

```ts
type AIPluginState = {
   aiMenuBlockID: undefined;
   aiMenuResponseStatus: 'initial'
} | {
   aiMenuBlockID: string;
   aiMenuResponseStatus: "generating" | "error" | "done";
 };
```

Reply:
good suggestion, see this commit: https://github.com/TypeCellOS/BlockNote/pull/980/commits/5a529cb49f4760e6ea56ea2179d9abfe824c7b8f